### DISCUSSION TITLE
Why does quasar recommend axios?

### DISCUSSION THREAD

**Discussion by yeus** [quasarframework/quasar#17309] [General - Components / Directives / etc]
In the documentation  there is this page:

https://quasar.dev/quasar-cli-vite/ajax-requests/

where it says ` We recommend selecting Axios during project initialization.`. I would like to know the reason for this. Doesn't "fetch" more or less do everything axios does. So its more a matter of taste? Except that axios makes your webpage larger? Or are there any other reasons for using axios isntead of the built-in fetch?  like when I do SSR (I haven't done SSR yet, so I don't know too much about that)?

**Comment by yusufkandemir** [comment]
You can use a search engine to find reasons, here is a Reddit post about this:
https://www.reddit.com/r/learnjavascript/comments/ymv7ex/why_do_people_use_axios_instead_of_fetch/

Personal thoughts and experience: I like the smaller benefits such as automatic JSON handling, throwing errors when an HTTP error happens, etc. However, configuring defaults such as a base URL, or being able to use interceptors such as one passing authorization header to all calls, are some of the more important features. I have worked on projects that had different APIs with different URLs and different authentication mechanisms, and Axios was really handy. If you don't need any of these and need to make only a few calls, then `fetch` would be just fine. I had projects like that in which I used `fetch`.
```ts
// a small example displaying the smaller benefits

// fetch
const response = await fetch('/test', {
  method: 'POST',
  body: JSON.stringify({
    foo: 1,
    bar: true,
  }),
});
if (!response.ok) {
  throw new Error('HTTP Error');
}
const data = (await response.json()) as { baz: string };

// axios
import axios from 'axios';

const { data } = await axios.post<{ baz: string }>('/test', {
  foo: 1,
  bar: true,
});
```

As for the documentation part, it was created in #3058, which is ~6 years old. Back in the day, things were different and there was also the need for IE11 support, which Axios had. So, it was a good recommendation. Today, it can be recommended, but not always the best solution.
The docs page has been updated to use Composition API, but the updates were pretty much limited to that. We could update the docs to reflect today's situation. We could also mention other suitable alternatives such as https://github.com/unjs/ofetch

**Reply by yeus** [reply]
ok, thank you @yusufkandemir , the #3058 explanation is all I needed :P. That makes sense. 

I was mainly wondering if there was anything specific to quasar with axios. E.g. missing support for fetch in SSR, or problems with some other cli mode. But that doesn't seem to be the case. I think updating the docs is a good idea!
